### PR TITLE
[CDAP-7465] Market place package view - Layout

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Streamlines the same component `MarketPlaceEntity` to be used across all tabs in marketplace
- Fixes the grid layout issue to be different columns in different widths (have the last package on its own line to not stretch the entire width)
- Adds `MarketPlaceUsecaseEntity` component to be used in usecase tab
- Adds `UsecaseTab` to be used in usecase in market place instead of `AllTab`
- Minor changes to the organization of components under Market.

### CODE REVIEW

File: cdap-ui/app/cdap/components/Market/AllTab/index.js [hunk 1]
```diff
@@ -16,43 +16,41 @@
...
+import {MyMarketApi} from 'api/market';
 require('./AllTabContents.less');
 
 export default class AllTabContents extends Component {
   constructor(props) {
     super(props);
     this.state = {
       searchStr: '',
-      entities: [],
+      entities: this.getFilterdEntities(),
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/Market/AllTab/index.js]: Filtered typo

File: cdap-ui/app/cdap/components/MarketPlaceUsecaseEntity/index.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+      );
+    }
+
+    return (
+      <div className="market-entity-actions">
+        {
+          this.state.entityDetail.actions.map((action, index) => {
+            let isCompletedAction = this.state.completedActions.indexOf(index) !== -1 ;
+            let actionName = T.translate('features.Market.action-types.' + action.type + '.name');
+            let actionIcon = iconMap[action.type];
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/MarketPlaceUsecaseEntity/index.js]: Can this be extracted to a helper function perhaps to improve readability?

**Comment by elfenheart** [comment]
👍 LGTM

**Comment by ajainarayanan** [comment]
@elfenheart Haven't added cursor pointer for each market place entity. I am adding it in the next PR (in case you notice it).

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**